### ISSUE TITLE
Excess property check doesn't work with deeply nested type intersection

### ISSUE THREAD

**Issue by xcrzx** [microsoft/TypeScript#47935]
# Bug Report

When using the intersection of deeply nested types (more than two levels) excess property check stops working. It was working as expected in `3.8`.

### 🔎 Search Terms

excess properties, extra properties, intersection

### 🕗 Version & Regression Information

- This changed between versions 3.8 and 3.9

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.5.4#code/KYDwDg9gTgLgBDAnmYcAqwDO8C8cDeAsAFBxwCGAXASWWQEYD812UAlgHYDmA3LXAF8+xAXABkNUhWpEpZAMYz+dOABMWMdt2Eqh-PSOEl5EDtjgcIAUShRo1DObyyyVSSsXuVa6gHIAjABMAMy+ADTKdABmEBB+MRDhcAD0yXBW4MDyMMCqFAjIqMC20HAAFsXA8bG+ahBYFhDwoGzmnAUocL74PnCsnLyCvpECEVKjJAbGpubFdlAOWLhebi50nmsq6l1BoWPecAnViWEpaTbzFJhwoCjZuSP7gk-05AtdrwBew4ZAA)

### 💻 Code

```ts
export type Test = {
  a: {
    b?: string;
  };
} & {
  a: {
    c: {
      d: string;
    };
  };
};

const noError: Test = {
  a: {
    c: {
      d: '123',
      foo: 'foo', // ❌ Expected a type error here: 'foo' does not exist in type '{ d: string; }'
    },
  },
};
```

### 🙁 Actual behavior

No type errors.

### 🙂 Expected behavior

I expect to see a type error:
```
Type '{ d: string; foo: string; }' is not assignable to type '{ d: string; }'.
  Object literal may only specify known properties, and 'foo' does not exist in type '{ d: string; }'.
```

Interesting enough that the error message shows correctly when I add an extra property somewhere else on an upper level:
```ts
const error: Test = {
  a: {
    c: {
      d: '123',
      foo: 'foo', // ✅ Error as expected: 'foo' does not exist in type '{ d: string; }'
    },
  },
  bar: 'baz' // <- added an extra prop here
};
```

**Comment by ahejlsberg** [comment]
This has been fixed since TS 5.0.
